src/scheduler/executionHistory.tsx

```diff
@@ -0,0 +1,219 @@
+
+  const [blueListDates, setBlueListDates] = useState<string[]>([]);
+  const [greyListDates, setGreyListDates] = useState<string[]>([]);
+  const [orangeListDates, setOrangeListDates] = useState<string[]>([]);
+  const [redListDates, setRedListDates] = useState<string[]>([]);
+  const [greenListDates, setGreenListDates] = useState<string[]>([]);
+  const [darkGreenListDates, setDarkGreenListDates] = useState<string[]>([]);
+
+  const handleMonthChange = () => {
+    setDagRunId('');
+    setSelectedDate(null);
+  };
+
+  const handleDateSelection = (selectedValue: any) => {
+    setDagRunId('');
+    setSelectedDate(selectedValue);
+  };
+
+  const CustomDay = (props: PickersDayProps<Dayjs>) => {
+    const { day, isFirstVisibleCell, isLastVisibleCell } = props;
+    if (isFirstVisibleCell) {
+      setStartDate(new Date(day.toDate()).toISOString());
+    }
+    if (isLastVisibleCell) {
+      const nextDate = new Date(day.toDate());
+      nextDate.setDate(day.toDate().getDate() + 1);
+      setEndDate(nextDate.toISOString());
+    }
+
+    const totalViewDates = day.date();
+
+    //Color codes to highlight dates in calendar 
+    //Blue color code for running status
+    const isBlueExecution =
+      blueListDates.length > 0 &&
+      blueListDates.includes(totalViewDates.toString().padStart(2, '0'));
+    //Grey color code for queued status
+    const isGreyExecution =
+      greyListDates.length > 0 &&
+      greyListDates.includes(totalViewDates.toString().padStart(2, '0'));
+    //Orange color code for combination of failed and success status
+    const isOrangeExecution =
+      orangeListDates.length > 0 &&
+      orangeListDates.includes(totalViewDates.toString().padStart(2, '0'));
+    //Red color code for only with failed status
+    const isRedExecution =
+      redListDates.length > 0 &&
+      redListDates.includes(totalViewDates.toString().padStart(2, '0'));
+    //Green color code for only one with success status
+    const isGreenExecution =
+      greenListDates.length > 0 &&
+      greenListDates.includes(totalViewDates.toString().padStart(2, '0'));
+    //Dark green color code for multiple with success status
+    const isDarkGreenExecution =
+      darkGreenListDates.length > 0 &&
+      darkGreenListDates.includes(totalViewDates.toString().padStart(2, '0'));
+
+    const isSelectedExecution = [selectedDate?.date()].includes(totalViewDates);
+
+    return (
+      <PickersDay
+        {...props}
+        style={{
+          border: isSelectedExecution
+            ? '3px solid var(--jp-ui-font-color0)'
+            : 'none',
+          borderRadius:
+            isGreenExecution ||
+            isRedExecution ||
+            isSelectedExecution ||
+            isOrangeExecution ||
+            isGreyExecution ||
+            isBlueExecution ||
+            isDarkGreenExecution
+              ? '50%'
+              : 'none',
+          backgroundColor: isDarkGreenExecution
+            ? '#188038'
```

Comment:
do these work in dark mode?

Reply:
Color codes for calendar works in both light and dark theme